src/components/TransactionHistoryRows/utils.ts

```diff
@@ -58,3 +63,49 @@ export const parseRelevantAssetFromTx = (txDetails: TxDetails, assetType: AssetT
       }
   }
 }
+
+type GetTradeFeesInput = {
+  buyAsset: Asset
+  sellAsset: Asset
+  buyAmount: string
+  sellAmount: string
+  blockTime: number
+  cryptoPriceHistoryData: PriceHistoryData
+}
+export const getTradeFees = memoize(
+  ({
+    sellAsset,
+    buyAsset,
+    buyAmount,
+    sellAmount,
+    blockTime,
+    cryptoPriceHistoryData,
+  }: GetTradeFeesInput) => {
+    const sellAssetPriceHistoryData = cryptoPriceHistoryData[sellAsset.assetId]
+    const buyAssetPriceHistoryData = cryptoPriceHistoryData[buyAsset.assetId]
+
+    if (!sellAssetPriceHistoryData || !buyAssetPriceHistoryData) return null
+
+    const sellAssetPriceAtDate = priceAtDate({
+      date: blockTime,
+      priceHistoryData: sellAssetPriceHistoryData,
+    })
+
+    const buyAssetPriceAtDate = priceAtDate({
+      date: blockTime,
+      priceHistoryData: buyAssetPriceHistoryData,
+    })
```

Comment:
if either of these are `'0'` return null and handle in consumer - it's possible we don't have data here.